src/components/Link.tsx

```diff
@@ -47,125 +47,131 @@ export interface IProps extends IBaseProps, LinkProps {
+    // this is to support the ButtonLink component which uses the `to` prop
+    let to = (toProp ?? href)!
+
+    const isDiscordInvite = url.isDiscordInvite(to)
+    if (isDiscordInvite) to = new URL(DISCORD_PATH, SITE_URL).href
+    const isExternal = url.isExternal(to)
+    const isHash = url.isHash(to)
+    const isStatic = url.isStatic(to)
+    const isPdf = url.isPdf(to)
+
+    const externalLinkEvent: MatomoEventOptions = {
+      eventCategory: `External link`,
+      eventAction: `Clicked`,
+      eventName: to,
+    }
+
+    const hashLinkEvent: MatomoEventOptions = {
+      eventCategory: `Hash link`,
+      eventAction: `Clicked`,
+      eventName: to,
+    }
+
+    const commonProps: LinkProps & { ref: React.ForwardedRef<any> } = {
+      ref,
+      href: to,
+      dir,
+      ...restProps,
+    }
+
+    // Must use Chakra's native <Link> for anchor links
+    // Otherwise the Gatsby <Link> functionality will navigate to homepage
+    // See https://github.com/gatsbyjs/gatsby/issues/21909
+    if (isHash) {
+      return (
+        <ChakraLink
+          onClick={(e) => {
+            // only track events on external links and hash links
+            if (!isHash) {
+              return
+            }
+
+            e.stopPropagation()
+            trackCustomEvent(
+              customEventOptions ? customEventOptions : hashLinkEvent
+            )
+          }}
+          {...commonProps}
+        >
+          {children}
+        </ChakraLink>
+      )
+    }
+
+    // Download link for internally hosted PDF's & static files (ex: whitepaper)
+    // Opens in separate window.
+    if (isExternal || isPdf || isStatic) {
+      return (
+        <ChakraLink
+          isExternal
+          onClick={(e) => {
+            // only track events on external links and hash links
+            if (!isExternal) {
+              return
+            }
+
+            e.stopPropagation()
+            trackCustomEvent(
+              customEventOptions ? customEventOptions : externalLinkEvent
+            )
+          }}
+          {...commonProps}
+        >
+          <>
+            {children}
+            <VisuallyHidden>(opens in a new tab)</VisuallyHidden>
+            {!hideArrow && (
+              <Icon
+                as={RxExternalLink}
+                boxSize="6"
+                p="1"
+                verticalAlign="middle"
+                me="-1"
+              />
+            )}
+          </>
+        </ChakraLink>
+      )
+    }
+
+    // Use `gatsby-theme-i18n` Link (which prepends lang path)
     return (
       <ChakraLink
-        href={to}
-        onClick={(e) => {
-          // only track events on external links and hash links
-          if (!isHash) {
-            return
-          }
-
-          e.stopPropagation()
-          trackCustomEvent(
-            customEventOptions ? customEventOptions : hashLinkEvent
-          )
-        }}
+        as={IntlLink}
```

Comment:
The Gatsby Link expects a `to` prop instead of the `href`.

Reply:
Ah shoot. I'll revert the addition of `href` to `commonProps` for now.